### ISSUE TITLE
FBeta binary average

### ISSUE THREAD

**Issue by zippeurfou** [Lightning-AI/metrics#530]
## 🚀 Feature

Have fbeta binary metric following sklearn

### Motivation

This is used and is the default in sklearn. It can be misleading if people use it without thinking.

### Code example
<img width="1713" alt="Screen Shot 2021-09-17 at 6 57 12 PM" src="https://user-images.githubusercontent.com/4520383/133862001-b7185540-0519-4160-a495-0a9dadad9de9.png">

```py
from sklearn.metrics import fbeta_score

y_true = torch.tensor([0,1,1,0,1,1])
y_pred_class = torch.tensor([1,0,1,0,1,0])

fbeta_score(y_true, y_pred_class, 1)

from torchmetrics import FBeta
f1 = FBeta(num_classes=2, beta=1)
f1(y_pred_class,y_true)

f1 = FBeta(num_classes=2, beta=1,average='none')
f1(y_pred_class,y_true)
```

**Comment by SkafteNicki** [comment]
You are right that the binary case of some of the classification metrics are not handle in a ideal way. Any suggestions what we could do to improve?

**Comment by zippeurfou** [comment]
Well sklearn binary = FBeta(num_classes=2, beta,average='none')[1]
So we could do a quick PR where we put the average by default to be binary the same way it is for sklearn and in the code just do average='none' and display only the positive one.

**Comment by Borda** [comment]
@zippeurfou mind send a quick PR? :rabbit:

**Comment by tridao** [comment]
I agree that this could be quite misleading for the default to diverge from sklearn (which I assume is most ppl are used to).

**Comment by SkafteNicki** [comment]
Issue will be fixed by classification refactor: see this issue https://github.com/Lightning-AI/metrics/issues/1001 and this PR https://github.com/Lightning-AI/metrics/pull/1195 for all changes

Small recap: This issue describe that metric `FBeta` metric is not the same as sklearn in the binary setting. The problem with the current implementation is that the metrics are calculated as average over the 0 and 1 class, which is wrong. After the refactor, using new the `bineary_fbeta_score` give the same result as sklearn:

```python
from torchmetrics.functional import binary_fbeta_score
from sklearn.metrics import fbeta_score
import torch

preds = torch.rand(10)
target = torch.randint(0, 2, (10,))
print(binary_fbeta_score(preds, target, beta=2.0) == fbeta_score(target, preds.round(), beta=2.0))  # True
```
sorry for the confusion this has created.
Issue will be closed when https://github.com/Lightning-AI/metrics/pull/1195 is merged.
